examples/ex9p.cpp

```diff
@@ -58,35 +58,130 @@ double inflow_function(const Vector &x);
 // Mesh bounding box
 Vector bb_min, bb_max;
 
+struct AIR_parameters
+{
+   int blocksize;
+   int distanceR;
+   std::string prerelax;
+   std::string postrelax;
+   int interp_type;
+   int relax_type;
+   int coarsen_type;
+   double strength_tolC;
+   double strength_tolR;
+   double filter_tolR;
+   double filterA_tol;
+};
+
+class AIR_prec : public Solver
```

Comment:
All of the `AIR_prec` class should be in an `#ifdef` for the hypre version.

Then you can remove the other `#ifdef`s inside this class.